__test__/saveSchedule.test.tsx

```diff
@@ -0,0 +1,149 @@
+import "@testing-library/jest-dom";
+import { screen, waitFor } from "@testing-library/react";
+import { jest } from "@jest/globals";
+
+import ScheduleUploadModal, { getDateTime } from "../components/ScheduleUploadModal";
+import ActionBar from "../components/ActionBar";
+import { setupUser, render, createMockToken, setupMockUserDB, mockToken } from "./util";
+import { userMajor, UserLogin, User } from "../services/user";
+
+jest.useFakeTimers({
+  doNotFake: [
+    "hrtime",
+    "nextTick",
+    "performance",
+    "queueMicrotask",
+    "requestAnimationFrame",
+    "cancelAnimationFrame",
+    "requestIdleCallback",
+    "cancelIdleCallback",
+    "setImmediate",
+    "clearImmediate",
+    "setInterval",
+    "clearInterval",
+    "setTimeout",
+    "clearTimeout"
+  ],
+  advanceTimers: false,
+  now: new Date("2020-01-01")
+});
+
+//  MOCKED JSON Data for the Courses
+const infoMocked = {
+  Major: userMajor()?.major.name,
+  Concentration: userMajor()?.concentration.name,
```

Comment:
These could be hard-coded values for the sake of having truly mocked data rather than pulling from the user service